### ISSUE TITLE
Filter Stage Type Dropdown by Application Providers

### ISSUE THREAD

**Issue by sbwsg** [spinnaker/spinnaker#2277]
*TL;DR*

Remove Kubernetes v1 stages from Stage Type dropdown if Kubernetes v2 is the only configured k8s provider for the current application.

*Cloud Provider*

GCP, Kubernetes

*Environment*

Ubuntu 14.04

*Feature Area*

Pipeline editing workflow

*Description*

When creating a new pipeline stage we have multiple different "Deploy" stages to choose from.  I'm attempting to push out a kubernetes deployment and so the correct stage is "Deploy (Manifest)".  However, I first attempted to use the plain "Deploy" stage, followed by clicking the "Add Server Group" button.  This caused errors in the console and an empty server group UI.  The reason for this UI state was not immediately obvious.

The problem appears to be that the UI is getting confused about v1 vs v2 k8s provider.  The v1 provider uses the Deploy stage and the v2 provider uses the Deploy (Manifest) stage.

*Steps to Reproduce*

1. Launch deck with kubernetes v2 provider enabled
2. Create an Application and configure it with the kubernetes provider only
3. Create a new pipeline.
4. Add a "Deploy" stage.
5. Click the "Add Server Group" button.
6. Notice that the UI becomes somewhat confused.
7. Try again and this time create a "Deploy (manifest)" stage instead.
8 Everything should work as expected.

*Additional Details*

Here are the two deploy stages as seen in the stage selection dropdown:

![deploy_deploy_manifest](https://user-images.githubusercontent.com/34253460/35457266-4a68dbf2-02a6-11e8-9496-df741b80b6bb.png)

Here's the empty form that I encountered after trying to Add Server Group in the Deploy stage:

![empty_form](https://user-images.githubusercontent.com/34253460/35457207-1d8ef6b6-02a6-11e8-9e34-efe5576dc928.png)

And the error message in the console:

![error_msg](https://user-images.githubusercontent.com/34253460/35457220-251131f6-02a6-11e8-9916-37db1b86ef73.png)

Will also need to filter the icons in the stage dropdown when multiple accounts are configured in an application.  In this screenshot the kubernetes icon shouldn't appear in the Deploy stage since the account configured in the application is a k8s v2 provider:

![before_icon_fix](https://user-images.githubusercontent.com/34253460/36054841-8507bdf2-0dc6-11e8-9a22-27134ccc5770.png)

And also filter out the provider choice modal:

![before_provider_choice_fix](https://user-images.githubusercontent.com/34253460/36055061-bd3e393e-0dc7-11e8-94bd-cc552c6d5641.png)

**Comment by sbwsg** [comment]
I didn't realise but it turns out we actually already do this.  Creating an application with only an appengine provider correctly hides the `Deploy (Manifest)` stage from the dropdown.

**Comment by lwander** [comment]
I don't think we do this correctly for v1 vs v2 of the k8s provider yet

**Comment by sbwsg** [comment]
Ahh ok that explains why I was seeing both in the description above.  I'll edit the issue to correctly reflect the problem.

**Comment by lwander** [comment]
Safe to close?
